cockpit/leapp.html

```diff
@@ -419,47 +410,105 @@ <h3 class="pull-left">
+                try {
+                    result = JSON.parse(String(result));
+                } catch(err) {
+                    return { success: false, errors: [{code: -1, err_message: String(err)}]};
+                }
+
+                if (result.errors !== undefined) {
+                    return { success: false, errors: result.errors };
+                }
+                else {
+                    var data = {};
+                    if (result.data !== undefined) {
+                        data = result.data;
+                    }
+
+                    return { success: true, data: data };
+                }
+            };
+
             return instance = {
                 run: function() {
                     if (proc == null) {
                         update_state('running');
-                        proc = cockpit.spawn(command, {
-                            directory: LEAPP_CONF["tool-workdir"],
-                            environ: ["SSH_AUTH_SOCK=" + AGENT_AUTH_SOCK],
-                            err: 'out',
-                            superuser: 'require',
-                        });
-                        proc.done(function(data) {
-                            update_state('success');
-                            process_output(data);
-                            output += data;
-                            write_log("Command completed successfully");
-                            instance.done = function(f) { f(output); };
-                            instance.then = function(f) { f(); };
-                            handlers.done.forEach(function(e) { e(output); });
-                            handlers.then.forEach(function(e) { e(output); });
-                        });
-                        proc.fail(function(err, data) {
+
+                        var fail = function() {
                             update_state('failed');
-                            output += data;
+
                             if (!options || options.log_stdout !== false) {
-                                process_output(data);
+                                process_output(output);
                             } else {
                                 process_output(output, true);
                             }
-                            write_log(`Command failed with status ${err.exit_status}!`);
+
                             instance.fail = function(f) { f(output); };
                             instance.then = function(f) { f(); };
                             handlers.fail.forEach(function(e) { e(output); });
                             handlers.then.forEach(function(e) { e(output); });
+                        }
+
+
+                        http = cockpit.http(LEAPP_CONF["daemon-port"], {
+                            address: LEAPP_CONF["daemon-address"],
+                        })
+                        request = http.post("/" + LEAPP_CONF["daemon-api-version"] + "/" + command[0], command[1], {"Keep-Alive": "300"})
```

Comment:
Keep-Alive - does it work? I mean golang http server is checking that?

Reply:
Hey, no it does not, I somehow forget to delete it.